src/test/store/receive-profile.test.js

```diff
@@ -503,6 +503,59 @@ describe('actions/receive-profile', function() {
     });
   });
 
+  describe('changeViewAndRecomputeProfileData', function() {
+    const browsingContextID = 123;
+    const innerWindowID = 111111;
+    function setup(initializeShowTabOnly: boolean = false) {
+      const store = blankStore();
+      const profile = getEmptyProfile();
+
+      profile.threads.push(
+        getEmptyThread({ name: 'GeckoMain', processType: 'tab', pid: 1 })
+      );
+
+      profile.meta.configuration = {
+        threads: [],
+        features: [],
+        capacity: 1000000,
+        activeBrowsingContextID: browsingContextID,
+      };
+      profile.pages = [
+        {
+          browsingContextID: browsingContextID,
+          innerWindowID: innerWindowID,
+          url: 'URL',
+          embedderInnerWindowID: 0,
+        },
+      ];
+
+      store.dispatch(viewProfile(profile));
+      if (initializeShowTabOnly) {
+        store.dispatch(changeViewAndRecomputeProfileData(browsingContextID));
+      }
+
+      return { ...store, profile };
+    }
+
+    it('should be able to switch to active tab view from the full view', function() {
+      const { dispatch, getState } = setup();
+      expect(UrlStateSelectors.getShowTabOnly(getState())).toBe(null);
+      dispatch(changeViewAndRecomputeProfileData(browsingContextID));
+      expect(UrlStateSelectors.getShowTabOnly(getState())).toBe(
```

Comment:
I think this behavior should be more aggressively tested. We should check that the profileView's data actually swaps out with new data.

Reply:
Yeah, you are right. I will add more tests after deciding the state architecture.